### PR TITLE
Fixes the lesson positions after a lesson edit page save

### PR DESCRIPTION
**Pull request by dmcavoy**
## Background
The relative position of lessons should be set based on their position within a script relative to other lockable/non-lockable lessons. When we moved the lockable setting to the lesson edit page for migrated courses relative position stopped getting updated when the lockable setting was changed. This is because the logic associated with updating relative position is part of the script saving process where the lockable setting used to live, in the script DSL.

## Solution
Adds a method to update the absolute and relative positions of all the lessons in a script after one of the lessons in that script is saved using the new lesson edit page. 

## Future Work
We are working toward a point where relative position will be based on if a lesson has a lesson plan not if its lockable but that will come later. 

## Links

- [lockable lessons and lessons without lesson plan spec](https://docs.google.com/document/d/1sAAVnwv3rXQEK-qa8JnkNXFfZ0vjWQyzJ9Yx_pG4bdg/edit#)
- [jira](https://codedotorg.atlassian.net/browse/PLAT-470)

## Testing story

Added a test to the lesson_controller_tests to check that update updates the positions of lessons correctly.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -1058,6 +1058,27 @@ def fix_script_level_positions
     end
   end
 
+  # Lessons unfortunately have 2 position values:
+  # 1. absolute_position: position within the Script
+  # 2. relative_position: position within the Script relative other lockable/non-lockable lessons
+  # This method updates the position values for all lessons in a script after
+  # a lesson is saved
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb]: I just want to point out that the lessons association orders the lessons by lesson group, and then within each lesson group the lessons are ordered by their absolute position:
* https://github.com/code-dot-org/code-dot-org/blob/9e64972c3c87b44f3fdaacdc55a7aa2c7ab70d66/dashboard/app/models/script.rb#L38-L39
* https://github.com/code-dot-org/code-dot-org/blob/7ec616278cdcb550a8ff85202bf5c9e53d442532/dashboard/app/models/lesson_group.rb#L23

I think this sounds right. Does that seem obvious enough, or do you think it's worth mentioning what the source of truth is here in the comments?

File: dashboard/test/controllers/lessons_controller_test.rb [hunk 2]
```diff
@@ -251,6 +255,26 @@ class LessonsControllerTest < ActionController::TestCase
...
+
+    put :update, params: @update_params
+
+    @lesson.reload
+    @lesson2.reload
+    assert_equal 1, @lesson.relative_position
+    assert_equal 1, @lesson.absolute_position
+    assert_equal 1, @lesson2.relative_position
+    assert_equal 2, @lesson2.absolute_position
+  end
```
Comment (davidsbailey) [inline dashboard/test/controllers/lessons_controller_test.rb:276]: nice test!

**Review by davidsbailey — APPROVED** [review]
Looks good!

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/models/script.rb]: Good call out. I can update the the comment to try to capture this.

**Review by bethanyaconnor — APPROVED** [review]

**Comment by dmcavoy** [comment]
Made a comment only update so going to merge ahead of test run.

**Pull request merged**